mscore/pagesettings.cpp

```diff
@@ -109,158 +111,211 @@ void PageSettings::setScore(Score* s)
-
-      twosided->setChecked(_twosided);
+      if (idxType < 0) {
+            QLocale local;
+            if (local.measurementSystem() == QLocale::MetricSystem)
+                  idxType = PAPER_TYPE_METRIC;
+            else
+                  idxType = PAPER_TYPE_IMPERIAL;
+      }            
+      typesList->setCurrentIndex(idxType); // typesList blocks signals - cleaner that way
+      typeChanged(idxType);                // typeChanged() loads sizesList
+      sizesList->setCurrentIndex(sizesList->findData(id));
+
+      if (odd->orientation() == QPageLayout::Portrait)
+            portraitButton ->setChecked(true);
+      else
+            landscapeButton->setChecked(true);
+
+      bool is2 = score->styleB(Sid::pageTwosided);
+      twosided->setChecked(is2);
+      for (auto w : { evenPageTopMargin, evenPageBottomMargin, evenPageLeftMargin, evenPageRightMargin })
+            w->setEnabled(is2);
+
+      ///!!!Minimum margins would be nice here, taken from the printer, maybe in main()...
+      QMarginsF marg = odd->margins();
+      oddPageTopMargin    ->setValue(marg.top());
+      oddPageBottomMargin ->setValue(marg.bottom());
+      oddPageLeftMargin   ->setValue(marg.left());
+      oddPageRightMargin  ->setValue(marg.right());
+
+      marg = even->margins();
+      evenPageTopMargin   ->setValue(marg.top());
+      evenPageBottomMargin->setValue(marg.bottom());
+      evenPageLeftMargin  ->setValue(marg.left());
+      evenPageRightMargin ->setValue(marg.right());
 
       pageOffsetEntry->setValue(score->pageNumberOffset() + 1);
-
       blockSignals(false);
+
+      // updateWidthHeight blocks signals for the width and height widgets
+      updateWidthHeight(odd->fullRect());
       }
 
 //---------------------------------------------------------
-//   inchClicked
+//   updateWidthHeight - updates Width and Height widgets
 //---------------------------------------------------------
 
-void PageSettings::inchClicked()
+void PageSettings::updateWidthHeight(const QRectF& rect)
       {
-      mmUnit = false;
-      updateValues();
+      pageWidth ->blockSignals(true);
+      pageHeight->blockSignals(true);
+      pageWidth ->setValue(rect.width());
+      pageHeight->setValue(rect.height());
+      pageWidth ->blockSignals(false);
+      pageHeight->blockSignals(false);
       }
 
 //---------------------------------------------------------
-//   mmClicked
+//   typeChanged - populates sizesList based on typesList
 //---------------------------------------------------------
 
-void PageSettings::mmClicked()
+void PageSettings::typeChanged(int idx)
       {
-      mmUnit = true;
-      updateValues();
+      switch (idx) {
+      case PAPER_TYPE_METRIC:
+            _sizes = &MScore::sizesMetric;
+            break;
+      case PAPER_TYPE_IMPERIAL:
+            _sizes = &MScore::sizesImperial;
+            break;
+      case PAPER_TYPE_OTHER:
+            _sizes = &MScore::sizesOther;
+            break;
+      }
```

Comment:
This brace and the entire body of the switch should be indented one more level.

Reply:
Fixed.  I was unsure about the switch indentation.  A JavaScript editor's auto-indent started me doing it that way.  I think it was Qt Creator with JavaScript, actually.